app/models/theme.rb

```diff
@@ -572,6 +572,8 @@ def update_setting(setting_name, new_value)
     raise Discourse::NotFound unless target_setting
 
     target_setting.value = new_value
+
+    Discourse.request_refresh! if target_setting.requests_refresh?
```

Comment:
Has the value actually been saved to the database at this point? To avoid race conditions (i.e. the client reloads before the assets are ready), I think we will need to make sure this runs after all of these things:

https://github.com/discourse/discourse/blob/bf34d3524b05e7f18cbdda11f357d11d642df3bb/app/models/theme.rb#L91-L96